modules/common/react/lib/OutsideEventListener.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+
+export interface Clickable {
+  onClick: Function;
+}
+
+export interface OutsideEventListenerProps {
+  children: JSX.Element;
+  onEscape?: (event?: KeyboardEvent) => void;
+  onOutsideClick?: (event?: MouseEvent) => void;
+}
+
+export class OutsideEventListener extends React.PureComponent<OutsideEventListenerProps> {
+  // we cannot use the DOM to detect if a click is outside when using portals,
+  // so we keep track of inside vs. outside click events
+  private insideClick = false;
+
+  public componentDidMount() {
+    document.addEventListener('click', this.onClick);
+    document.addEventListener('keydown', this.onKeyDown);
```

Comment:
Interesting. I'm interested in this for Modal accessibility, but should clicks and escape keys be managed together?